kernel/thread.c

```diff
@@ -1702,7 +1838,8 @@ void myst_wait_on_child_processes(myst_process_t* process)
         p = process->prev_process;
         while (p)
         {
-            if (p->ppid == pid)
+            if (is_main_thread ||
+                ((p->ppid == pid) && (p->is_parent_of_pseudo_fork_process)))
```

Comment:
Not sure I understand the `(p->is_parent_of_pseudo_fork_process)` part of the expression.